extensions/chocolatey-fosshub.extension/extensions/Get-UrlFromFosshub.psm1

```diff
@@ -1,28 +1,48 @@
-# Get the resolved URL form a FossHub download link.
+# Get the resolved URL from a FossHub download link.
 #
-# Takes a FossHub URL in the “genLink” format and returns the generated
+# Takes a FossHub URL and returns the generated
 # expiring download link for the file.
 #
 # Usage: Get-UrlFromFosshub genLink_url
 # Example:
-# Get-UrlFromFosshub http://www.fosshub.com/genLink/Data-Crow/FILE.zip
+# Get-UrlFromFosshub https://www.fosshub.com/Audacity.html/audacity-win-2.1.2.exe
 
-Function Get-UrlFromFosshub($genLinkUrl) {
+Function Get-UrlFromFosshub($linkUrl) {
 
-  $fosshubAppName = $genLinkUrl -match 'genLink/(.+?)/'
+  $fosshubAppName = $linkUrl -match 'www.fosshub.com/(.*)/(.*)'
   # If there’s no match, it means that it’s not a FossHub genLink URL.
   # Then this function simply returns the input URL.
   if (!$Matches) {
-    return $genLinkUrl
+    return $linkUrl
   }
+
   # Get the actual FossHub appname from the matches array
+  # From the example above, this would be Audacity.html
   $fosshubAppName = $Matches[1]
 
-  $referer = "http://www.fosshub.com/${fosshubAppName}.html"
+  # Get the file name from the URL
+  # From the example above, this would be audacity-win-2.1.2.exe
+  $fosshubFileName = $Matches[2]
+
+  # Create the contents of the href that is being searched for
+  # From the example above, this would be /Audacity.html/audacity-win-2.1.2.exe
+  $hrefText = "/$fosshubAppName/$fosshubFileName"
+
+  # Construct the Regular Expression pattern.  An example of what is being searched for is this
+  # <a href="/Audacity.html/audacity-win-2.1.2.exe" rel="nofollow" class="dwl-link xlink" file="audacity-win-2.1.2.exe" data="https://download.fosshub.com/Protected/expiretime=1475696791;badurl=aHR0cDovL3d3dy5mb3NzaHViLmNvbS9BdWRhY2l0eS5odG1s/606878a89d0b1e2bf998e0482bf081a451e8d98479d8b246a83d33db1a597538/Audacity/audacity-win-2.1.2.exe">
+  $regexPattern = "<a href=`"$hrefText`".*data=`"(.*)`""
+
+  $referer = "https://www.fosshub.com/${fosshubAppName}"
   $webClient = New-Object System.Net.WebClient
   $webClient.Headers.Add("Referer", $referer)

```

Comment:
I didn't have to alter referer with smplayer, that was true before....

Anyway, this is big issue IMO. People use WebClient all the time because helpers are not complete. Implementing better helper should be done so that people can stop using other methods.